src/porepy/models/energy_balance.py

```diff
@@ -673,30 +739,47 @@ def bc_values_enthalpy_flux(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray
             Array with boundary values for the enthalpy.
 
         """
-        # List for all subdomains
-        bc_values = []
-
-        # Loop over subdomains to collect boundary values
-        for sd in subdomains:
-            vals = np.zeros(sd.num_faces)
-            # If you know the boundary temperature, you might want to reconstruct
-            # the enthalpy flux at the boundary. Using Dirichlet BCs, the value assigned
-            # herein will be multiplied with the Darcy flux -K \nabla p. Thus, one would
-            # normally convert to fluid flux multiplying by (density/viscosity). Then,
-            # if using EnthalpyFromTemperature, one would compute enthalpy value by
-            # multiplying by specific heat capacity and
-            # (boundary temperature - reference temperature). Note that using
-            # non-default constitutive laws in the interior might require changes to the
-            # above recipe.
-
-            #  Append to list of boundary values
-            bc_values.append(vals)
-
-        # Concatenate to single array and wrap as ad.DenseArray
-        bc_values_ad = pp.wrap_as_ad_array(
-            np.hstack(bc_values), name="bc_values_enthalpy"
+
+        boundary_projection = pp.ad.BoundaryProjection(self.mdg, subdomains=subdomains)
+        boundary_grids = self.subdomains_to_boundary_grids(subdomains)
+
+        enthalpy_dirichlet = self.fluid_enthalpy(boundary_grids) * self.fluid_density(
+            boundary_grids
+        )
+        enthalpy_neumann = self.enthalpy_flux(boundary_grids)
+
+        result = enthalpy_dirichlet + enthalpy_neumann
+        result = boundary_projection.boundary_to_subdomain @ result
+        result.set_name("bc_values_enthalpy")
+        return result
+
+    def boundary_temperature(self, boundary_grid: pp.BoundaryGrid) -> np.ndarray:
+        """TODO"""
+        return self.fluid.temperature() * np.ones(boundary_grid.num_cells)
+
+    def boundary_fourier_flux(self, boundary_grid: pp.BoundaryGrid) -> np.ndarray:
+        """TODO"""
+        return np.zeros(boundary_grid.num_cells)
```

Comment:
With boundary grids, this is not a problem anymore. Previously, it was a problem, because you had to set values for Dirichlet enthalpy flux. Now this is done by the ad machinery.